crates/solidity/src/zksolc/main.rs

```diff
@@ -190,18 +194,36 @@ fn main_inner() -> anyhow::Result<()> {
         );
     } else if arguments.output_assembly || arguments.output_binary {
         for (path, contract) in build.contracts.into_iter() {
+            let bytescode = contract.build.bytecode;
+
             if arguments.output_assembly {
-                println!(
-                    "Contract `{}` assembly:\n\n{}",
-                    path, contract.build.assembly_text
-                );
+                let program_blob = match ProgramBlob::parse(bytescode.as_slice()) {
+                    Ok(blob) => blob,
+                    Err(error) => {
+                        bail!("Failed to parse program blob: {}", error);
+                    }
+                };
+
+                let disassembler_object =
```

Comment:
Another nitpick, in rust land we would name this variable just `disassembler` :upside_down_face:

Reply:
ok